catalogue/webapp/components/ConfirmItemRequest/ConfirmItemRequest.tsx

```diff
@@ -184,37 +202,91 @@ const ErrorDialog: FunctionComponent<ErrorDialogProps> = ({ setIsActive }) => (
   </>
 );
 
+type RequestingState = null | 'requesting' | 'confirmed' | 'error';
+
 const ConfirmItemRequest: FunctionComponent<Props> = props => {
   const openButtonRef = useRef<HTMLButtonElement>(null);
-  const { item, work, setIsActive, ...modalProps } = props;
-  const [isConfirmed, setIsConfirmed] = useState(false);
-  const [isLoading, setIsLoading] = useState(false);
-  const [isError, setIsError] = useState(false); // TODO: implement this if something goes wrong with API call
-
+  const { item, work, setIsActive, user, initialHoldNumber, ...modalProps } =
+    props;
+  const [requestingState, setRequestingState] = useState<RequestingState>(null);
```

Comment:
Why `null`?

Reply:
just being explicit that there is no value, as it this point no request has been attempted

Reply:
This is inferred if you leave the type empty: